skimage/io/_plugins/tifffile_plugin.py

```diff
@@ -24,7 +24,16 @@ def imread(fname, dtype=None, **kwargs):
     .. [1] http://www.lfd.uci.edu/~gohlke/code/tifffile.py
 
     """
+
     if 'img_num' in kwargs:
         kwargs['key'] = kwargs.pop('img_num')
-    with TiffFile(fname) as tif:
+
+    # separate TiffFile kwargs from other kwargs
+    tiff_keys = ['multifile', 'multifile_close', 'pages', 'fastij', 'is_ome']
+    kwargs_tiff = parse_kwargs(kwargs, *tiff_keys)
+    for key in tiff_keys:
```

Comment:
Why is this necessary?

I'm inclined to omit the need to call `parse_kwargs` unless we want to pass other kwargs to a different function. 

My reasoning is that 
1. It adds unecessary overhead.
2. We don't actually test this functionality in the test suite. (i.e. the test suite as is will likely pass if we remove these 3 lines)
3. By parsing the kwargs, we don't issue any error messages if a user provides a faulty kwargs.

Thoughts?

Reply:
> Why is this necessary?
> I'm inclined to omit the need to call `parse_kwargs` unless we want to pass other kwargs to a different function.

Exactly, it's to separate the arguments passed to `TiffFile(fname, **kwargs_tiff)` and `tif.asarray(**kwargs)`.

`parse_kwargs` should remove the keys from dict, so I'm not sure why `for key in tiff_keys: kwargs.pop(key, None)` is necessary.

Reply:
@hmaarrfk @cgohlke I feel like you are saying different things? @hmaarrfk are you saying that we can just pass all kwargs unmodified to `TiffFile` and then to `tif.asarray`? That seems like poor practice to me.

However, if `parse_kwargs` modifies the dictionary in place as @cgohlke suggests, then, sure, let's remove the for-loop, but let's add a comment above the `parse_kwargs` line that records this, as in-place modifications are not obvious on reading the code.

Reply:
```suggestion
```

Reply:
Oh i get it. From the docstring

```
Return dict with keys from keys|keyvals and values from kwargs|keyvals.

    Existing keys are deleted from kwargs.
```

@MarcelBeining please remove the `kwargs.pop` construct

Reply:
> just pass all kwargs unmodified to `TiffFile` and then to `tif.asarray`

I think that will fail for non-empty `kwargs`.

Reply:
> if `parse_kwargs` modifies the dictionary in place

I know, a little un-intuitive, but it's a useful helper function in my packages...

```
>>> help(parse_kwargs)
Help on function parse_kwargs in module skimage.external.tifffile.tifffile:

parse_kwargs(kwargs, *keys, **keyvalues)
    Return dict with keys from keys|keyvals and values from kwargs|keyvals.

    Existing keys are deleted from kwargs.

    >>> kwargs = {'one': 1, 'two': 2, 'four': 4}
    >>> kwargs2 = parse_kwargs(kwargs, 'two', 'three', four=None, five=5)
    >>> kwargs == {'one': 1}
    True
    >>> kwargs2 == {'two': 2, 'four': 4, 'five': 5}
    True
```

Reply:
Yeah,  I've seen a few people create these helper functions that would split dictionaries into two. I think it is fine to use it here as it part of the tifffile plugin.